app/assets/javascripts/comfortable_mexican_sofa/admin/modules/EmbeddedGoogleAnalytics.js

```diff
@@ -0,0 +1,173 @@
+define(['jquery', 'DoughBaseComponent'], function($, DoughBaseComponent) {
+  'use strict';
+
+  var defaultConfig = { };
+
+  function EmbeddedGoogleAnalytics($el, config) {
+    EmbeddedGoogleAnalytics.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  /**
+   * Inherit from base module, for shared methods and interface
+   */
+  DoughBaseComponent.extend(EmbeddedGoogleAnalytics);
+
+  EmbeddedGoogleAnalytics.componentName = 'EmbeddedGoogleAnalytics';
+
+  EmbeddedGoogleAnalytics.prototype.init = function(initialised) {
+    var path = this.config['path'];
+
+    (function(w,d,s,g,js,fs){
+        g=w.gapi||(w.gapi={});g.analytics={q:[],ready:function(f){this.q.push(f);}};
+          js=d.createElement(s);fs=d.getElementsByTagName(s)[0];
+            js.src='https://apis.google.com/js/platform.js';
+              fs.parentNode.insertBefore(js,fs);js.onload=function(){g.load('analytics');};
+    }(window,document,'script'));
+
+    gapi.analytics.ready(function() {
+      /**
+       * Authorize the user immediately if the user has already granted access.
+       * If no access has been created, render an authorize button inside the
+       * element with the ID "embed-api-auth-container".
+       */
+      gapi.analytics.auth.authorize({
+        container: 'embed-api-auth-container',
+        clientid: '46744299348-7t8tn92mh4pv01vk89llfv0u3q7kup2j.apps.googleusercontent.com',
+      });
+
+      var dataChart1 = new gapi.analytics.googleCharts.DataChart({
```

Comment:
Unless I'm mistaken these all seem to be doing the same thing except targetting a different container? Perhaps a method with that as an arg.

That method can perhaps return the output of this call, so you could have here:

```js
var dataChart1 = this.createDataChart('chart-1-container');
```